kernel/src/net/socket/netlink/common/mod.rs

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::{AtomicBool, Ordering};
+
+pub use bound::BoundNetlink;
+pub(super) use unbound::UnboundNetlink;
+
+use super::{GroupIdSet, NetlinkSocketAddr};
+use crate::{
+    events::IoEvents,
+    match_sock_option_ref,
+    net::socket::{
+        netlink::{table::SupportedNetlinkProtocol, AddMembership, DropMembership},
+        options::SocketOption,
+        private::SocketPrivate,
+        util::datagram_common::{self, select_remote_and_bind, Bound, Inner},
+        MessageHeader, SendRecvFlags, Socket, SocketAddr,
+    },
+    prelude::*,
+    process::signal::{PollHandle, Pollable, Pollee},
+    util::{MultiRead, MultiWrite},
+};
+
+mod bound;
+mod unbound;
+
+pub struct NetlinkSocket<P: SupportedNetlinkProtocol> {
+    inner: RwMutex<Inner<UnboundNetlink<P>, BoundNetlink<P::Message>>>,
+
+    is_nonblocking: AtomicBool,
+    pollee: Pollee,
+}
+
+pub trait UnboundNetlinkCommon<Message: 'static> = datagram_common::Unbound<
+    Endpoint = NetlinkSocketAddr,
+    Bound = BoundNetlink<Message>,
+    BindOptions = (),
+>;
+pub trait BoundNetlinkCommon = datagram_common::Bound<Endpoint = NetlinkSocketAddr>;
```

Comment:
`BoundNetlinkCommon` and `UnboundNetlinkCommon` are internal traits. They're only used in this single file. I don't think they need to be `pub`.

Reply:
Besides, `UnboundNetlink<P>: UnboundNetlinkCommon<P::Message>` seems already satisfied? Or am I missing something?

```rust
impl<P: SupportedNetlinkProtocol> datagram_common::Unbound for UnboundNetlink<P> {
```

Reply:
Oh, yes. This bound `UnboundNetlink<P>: UnboundNetlinkCommon<P::Message>` can be removed. I also removed the trait alias of `BoundNetlinkCommon` now since it's short.